Model/Sync/Catalog/Services/ProductsSyncService.php

```diff
@@ -68,6 +77,31 @@ public function getProductIdsFromSyncTableByStoreIdAndParentYotpoId($storeId, $p
         return array_keys($items);
     }
 
+    /**
+     * @param integer $productId
+     * @return bool
+     */
+    public function checkProductShouldBeSyncedByAttribute($productId)
```

Comment:
Why not use batch with IN clause?
It's very similar to the `getCollectionForSync` method.

Reply:
It was O(1) complexity, changed it anyway to use IN clause, as it was a small change to make.